File: backend/service-on-wheels-backend/src/main/java/com/serviceonwheels/auth_service/service/NotificationService.java
package com.serviceonwheels.auth_service.service;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class NotificationService {

    public void notifyMechanicAssigned(String requestId) {
        log.debug("[Notification] Mechanic assigned for request {}", requestId);
    }

    public void notifyMechanicArriving(String requestId) {
        log.debug("[Notification] Mechanic arriving for request {}", requestId);
    }

    public void notifyServiceCompleted(String requestId) {
        log.debug("[Notification] Service completed for request {}", requestId);
    }
}

File: customer-app/src/app/services/notification.service.ts
import { Injectable, signal } from '@angular/core';

export type NotificationEventType = 'mechanic-assigned' | 'mechanic-arriving' | 'service-completed';

export interface NotificationEvent {
  type: NotificationEventType;
  title: string;
  message: string;
  createdAt: string;
}

@Injectable({ providedIn: 'root' })
export class NotificationService {
  readonly events = signal<NotificationEvent[]>([]);

  push(event: NotificationEvent): void {
    this.events.update((items) => [event, ...items]);
  }

  notifyMechanicAssigned(requestId: string): void {
    this.push({
      type: 'mechanic-assigned',
      title: 'Mechanic Assigned',
      message: `A mechanic is on the way for request ${requestId}.`,
      createdAt: new Date().toISOString(),
    });
  }

  notifyMechanicArriving(requestId: string): void {
    this.push({
      type: 'mechanic-arriving',
      title: 'Mechanic Arriving',
      message: `Your mechanic is arriving at your location for request ${requestId}.`,
      createdAt: new Date().toISOString(),
    });
  }

  notifyServiceCompleted(requestId: string): void {
    this.push({
      type: 'service-completed',
      title: 'Service Completed',
      message: `Service request ${requestId} is complete.`,
      createdAt: new Date().toISOString(),
    });
  }
}
